File: src/ruby/Busca_Sequencial_Recursiva.rb
require "test/unit/assertions"
include Test::Unit::Assertions

# Retorna o índice do elemento. Se não existir, retorna -1.

def busca_sequencial_recursiva(lista, valor, index=0)
    if lista[index] == valor
        return index
    elsif index == lista.size
        return -1
    else
        index +=1
        busca_sequencial_recursiva(lista, valor, index)
    end
end

assert_equal busca_sequencial_recursiva([1, 2, 3, 4, 5], 1, index=0), 0
assert_equal busca_sequencial_recursiva(["a", "b", "c"], 1, index=0), -1
assert_equal busca_sequencial_recursiva([10, 9, 7, 2, 4, 5, 6, 99, 11, 4], 7, index=0), 2

File: src/c/BuscaSequencialRecursiva.c
#include <stdio.h>

int buscaSequencialRecursiva(int vetor[], int i, int buscado) {
    size_t n = sizeof(vetor) / sizeof(vetor[0]);
    if (vetor[i] == buscado)
    {
        return i;
    } 
    else if (i == n)
    {
        return -1;
    }
    else
    {
        buscaSequencialRecursiva(vetor, i+1, buscado);
    }
}

int main(){
    int a[] = {1, 2, 3, 4, 5, 6, 7, 8};
    printf("Valor %d no índice %d\n", 1, buscaSequencialRecursiva(a, 0, 1));
    buscaSequencialRecursiva(a, 0, 10);
}